src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/WorkspaceDiagnosticsRefresher.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Hosting;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Protocol;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer;
+
+internal class WorkspaceDiagnosticsRefresher : IRazorStartupService
```

Comment:
Like `WorkspaceSemanticTokensRefreshNotifier`, should this be disposable and have a `CancellationTokenSource` to cancel the refresh task? It seems to me that this could potentially send notifications after the language server has been shut down.

Reply:
FWIW, without disposal, you might see random failures in unit tests if it tries to send notifications after the test has stopped completed.